### ISSUE TITLE
Question about how to activate the slirp module.

### ISSUE THREAD

**Issue by ydonghyuk** [OP-TEE/optee_os#5869]
This is a question about how to activate the slirp module.

When running qemu, the following error occurs.
"network backend 'user' is not compiled into this binary"

If anyone can help, please.

**Comment by jforissier** [comment]
`sudo apt install libslirp-dev`

https://optee.readthedocs.io/en/latest/building/prerequisites.html

**Comment by ydonghyuk** [comment]
@jforissier 

I know it doesn't support Ubuntu 18.04.

I can't afford to upgrade Ubuntu.
Is the libslirp-dev package still present in 18.04?

If you know of a possible way in 18.04, please let me know.

**Comment by jforissier** [comment]
> @jforissier
> 
> I know it doesn't support Ubuntu 18.04.
> 
> I can't afford to upgrade Ubuntu. Is the libslirp-dev package still present in 18.04?

Apparently not: https://packages.ubuntu.com/search?keywords=libslirp-dev&searchon=names

> If you know of a possible way in 18.04, please let me know.

Perhaps compile and install it from https://gitlab.freedesktop.org/slirp/libslirp.

**Comment by ydonghyuk** [comment]
@jforissier 

Thanks for that info.
I will try and share.

Is there anything that needs to be modified when compiling qemu after installing libslirp-dev? (e.g. --enable-slirp)

**Comment by jforissier** [comment]
> Is there anything that needs to be modified when compiling qemu after installing libslirp-dev? (e.g. --enable-slirp)

No, I think the QEMU configure script should detect that libslirp is available automatically (whether it came with libslirp-dev or was installed from the Git should not make a difference hopefully).

**Comment by ydonghyuk** [comment]
@jforissier 

Successfully qemu booted.

Thank you sincerely.
